### PR TITLE
ompBLAS functions

### PR DESCRIPTION
**Pull request by kgasperich**
## Proposed changes

modified `ompBLAS::gemv` and `ompBLAS::gemv_batched` to allow `trans = 'N'` (tests added)
added tests for `ompBLAS::ger` and `ompBLAS::ger_batched`
added `ompBLAS::copy_batched` and `ompBLAS::copy_batched_offset` (tests not yet added)

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New feature
- Testing changes (e.g. new unit/integration/performance tests)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
JLSE

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- No*. Code added or changed in the PR has been clang-formatted (BLAS argument names are [vertically aligned](https://github.com/kgasperich/qmcpack/blob/fa2a39c8424cb52ef744e30a1c3347f499c96096/src/Platforms/OMPTarget/ompBLAS.cpp#L468-L474), but everything else has been clang-formatted)
- No*. This PR adds tests to cover any new code, or to catch a bug that is being fixed (gemv is covered, copy is not)
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/Platforms/OMPTarget/ompBLAS.cpp [hunk 1]
```diff
@@ -411,5 +442,150 @@ ompBLAS_status ger_batched(ompBLAS_handle&                   handle,
...
+                                 const T* const  x[],
+                                 const int       incx,
+                                 T* const        y[],
+                                 const int       incy,
+                                 const int       batch_count)
+{
+  if (batch_count == 0) return 0;
+
+  //if (incx !=1 || incy != 1)
+  //  throw std::runtime_error("incx !=1 or incy != 1 are not implemented in ompBLAS::copy_batched_impl!");
```
Comment (ye-luo) [inline src/Platforms/OMPTarget/ompBLAS.cpp:459]: Remove?

File: src/Platforms/OMPTarget/ompBLAS.cpp [hunk 2]
```diff
@@ -411,5 +442,150 @@ ompBLAS_status ger_batched(ompBLAS_handle&                   handle,
...
+                                        const int       incx,
+                                        T* const        y[],
+                                        const int       y_offset,
+                                        const int       incy,
+                                        const int       batch_count)
+{
+  if (batch_count == 0) return 0;
+
+  //if (incx !=1 || incy != 1)
+  //  throw std::runtime_error("incx !=1 or incy != 1 are not implemented in ompBLAS::copy_batched_impl!");
```
Comment (ye-luo) [inline src/Platforms/OMPTarget/ompBLAS.cpp:528]: Please remove commented code.

File: src/Platforms/OMPTarget/ompBLAS.cpp [hunk 3]
```diff
@@ -411,5 +442,150 @@ ompBLAS_status ger_batched(ompBLAS_handle&                   handle,
...
+
+  PRAGMA_OFFLOAD("omp target teams distribute parallel for collapse(2) is_device_ptr(x, y)")
+  for (size_t ib = 0; ib < batch_count; ib++)
+    for (size_t i = 0; i < n; i++)
+      y[ib][y_offset + i * incy] = x[ib][x_offset + i * incx];
+  return 0;
+}
+
+ompBLAS_status copy_batched_offset(ompBLAS_handle&    handle,
```
Comment (ye-luo) [inline src/Platforms/OMPTarget/ompBLAS.cpp:537]: This is an extension of BLAS. So please document this function especially its arguments.

File: src/Platforms/OMPTarget/ompBLAS.cpp (hunk 1 above)
Comment (kgasperich) [reply src/Platforms/OMPTarget/ompBLAS.cpp]: fixed

File: src/Platforms/OMPTarget/ompBLAS.cpp (hunk 2 above)
Comment (kgasperich) [reply src/Platforms/OMPTarget/ompBLAS.cpp]: fixed

File: src/Platforms/OMPTarget/ompBLAS.cpp (hunk 3 above)
Comment (kgasperich) [reply src/Platforms/OMPTarget/ompBLAS.cpp:531]: I added some documentation in ompBLAS.hpp

**Comment by ye-luo** [comment]
Test this please

**Review by ye-luo — APPROVED** [review]

**Review by prckent — APPROVED** [review]
LGTM

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**